@here/harp-mapview/lib/poi/PoiManager.ts

```diff
@@ -458,6 +411,85 @@ export class PoiManager {
         }
     }
 
+    /**
+     * Create and add POI [[TextElement]]s to tile with a series of positions.
+     */
+    private addPoi(
+        tile: Tile,
+        poiGeometry: PoiGeometry,
+        technique: PoiTechnique,
+        techniqueIdx: number,
+        positions: THREE.BufferAttribute
+    ) {
+        const poiTechnique = technique as PoiTechnique;
+        let imageTextureName = poiTechnique.imageTexture;
+        if (typeof poiTechnique.imageTexturePrefix === "string") {
+            imageTextureName = poiTechnique.imageTexturePrefix + imageTextureName;
+        }
+        if (typeof poiTechnique.imageTexturePostfix === "string") {
+            imageTextureName = imageTextureName + poiTechnique.imageTexturePostfix;
+        }
+
+        if (poiGeometry.stringCatalog === undefined) {
+            return;
+        }
+        const poiTableName = poiTechnique.poiTable;
+        let poiName = poiTechnique.poiName;
+
+        for (let i = 0; i < positions.count; ++i) {
+            const x = positions.getX(i);
+            const y = positions.getY(i);
+            const z = positions.getZ(i);
+
+            assert(poiGeometry.texts.length > i);
+            let text = poiGeometry.stringCatalog[poiGeometry.texts[i]];
```

Comment:
Why not just?
`const text = poiGeometry.stringCatalog[poiGeometry.texts[i]] || "";`

Reply:
Don't ask me ... not my code :).

Reply:
@sasha240100, @kkostecki 

While i agree that `|| ""` is more readable and correct in this case, please note that we have very strict rule disallowing any truthy/falsy statements an this one is exactly false`y for `""` in array albeit correct ("" yields "", so no harm) .

That's why `||` and `&&` on non-booleans is avoided like a plague even if we loose some compactness of the code.